src/rinterface_extra.c

```diff
@@ -2429,12 +2435,16 @@ void R_igraph_error_handler(const char *reason, const char *file,
    * IGRAPH_FINALLY_FREE() because 'reason' might be allocated on the heap and
    * IGRAPH_FINALLY_FREE() can then clean it up. */
 
-  if (R_igraph_errors_count == 0) {
+  if (R_igraph_errors_count == 0 || R_igraph_in_r_check != 0) {
```

Comment:
@krlmlr shouldn't be here `if (R_igraph_errors_count == 0 || R_igraph_in_r_check == 0) {` ?

`0` means that error was accrued not in context of `R_IGRAPH_CHECK` and we should print error anyway.

Reply:
Good catch!